### PR TITLE
Add firebase, Jwt , AuthService, Google provider

### PR DESCRIPTION
**Pull request by SayPls**
Code Rewie

### CODE REVIEW

File: backend/Codi.Core/Codi.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -40,5 +44,28 @@ public static void AddCodiCoreContext(this IServiceCollection services, IConfigu
...
                     opt => opt.MigrationsAssembly(typeof(CodiCoreContext).Assembly.GetName().Name)));
         }
+
+        public static void ServiceJwtFirebase(this IServiceCollection services, IConfiguration configuration)
+        {
+            var pathToKey = Path.Combine(Directory.GetCurrentDirectory(), "keys", "firebase_admin_sdk.json");
+            FirebaseApp.Create(new AppOptions
+            {
+                Credential = GoogleCredential.FromFile(pathToKey)
+            });
```
Comment (tatianahutii) [inline backend/Codi.Core/Codi.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs:54]: Why do we need this?

File: backend/Codi.Core/Codi.Core.WebAPI/Properties/launchSettings.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
       "environmentVariables": {
         "ASPNETCORE_ENVIRONMENT": "Development"
       },
-      "applicationUrl": "https://localhost:7011;http://localhost:5011",
+      "applicationUrl": "https://localhost:7011;http://localhost:5050",
```
Comment (tatianahutii) [inline backend/Codi.Core/Codi.Core.WebAPI/Properties/launchSettings.json:19]: Why did you change it?

File: frontend/.firebaserc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (tatianahutii) [inline frontend/.firebaserc:1]: Why do we need this?

File: frontend/angular.json [hunk 4]
```diff
@@ -124,9 +124,19 @@
...
+        },
+        "deploy": {
+          "builder": "@angular/fire:deploy",
+          "options": {
+            "prerender": false,
+            "ssr": false,
+            "browserTarget": "codi:build:production",
+            "firebaseProject": "codi-9ff49",
+            "firebaseHostingSite": "codi-9ff49"
+          }
```
Comment (tatianahutii) [inline frontend/angular.json:136]: Please remove this

File: frontend/firebase.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (tatianahutii) [inline frontend/firebase.json:1]: Why do we need this?

File: frontend/src/app/core/interceptors/auth.interceptor.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthInterceptor implements HttpInterceptor {
+  constructor(private authService: AuthService) {
+
+  }
+
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    let token = this.authService.getToken();
```
Comment (tatianahutii) [inline frontend/src/app/core/interceptors/auth.interceptor.ts:15]: You need to get a token only one time, not for every request

File: frontend/src/app/core/models/current-user.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export class CurrentUser {
+  isSignedIn: boolean;
+  displayName: string | null;
+  email: string | null;
```
Comment (tatianahutii) [inline frontend/src/app/core/models/current-user.ts:4]: Make these fields optional

File: frontend/src/app/core/services/auth.service.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+      }
+    })
+
+  }
+
+  configureAuthState(firebaseUser: firebase.User) {
+    if(firebaseUser) {
+      firebaseUser.getIdToken().then((theToken) => {
+        console.log('we have a token');
+        this.http.post('http://localhosts:5050/token/verify', {token: theToken})
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts:33]: It will not work on production

File: frontend/src/app/core/services/auth.service.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    } else {
+      this.doSignedOutUser();
+    }
+  }
+
+  private doSignedOutUser() {
+    let theUser = new CurrentUser();
+    theUser.displayName = null;
+    theUser.email = null;
+    theUser.isSignedIn = false;
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts:58]: Why do we need this?

File: backend/Codi.Core/Codi.Core.WebAPI/Codi.Core.WebAPI.csproj [hunk 10]
```diff
@@ -30,4 +31,8 @@
     <ProjectReference Include="..\Codi.Core.BLL\Codi.Core.BLL.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <_ContentIncludedByDefault Remove="keys\firebase_admin_sdk.json" />
+  </ItemGroup>
+
```
Comment (tatianahutii) [inline backend/Codi.Core/Codi.Core.WebAPI/Codi.Core.WebAPI.csproj:37]: Do we need this?

File: frontend/.gitignore [hunk 11]
```diff
@@ -40,3 +40,8 @@ testem.log
 # System files
 .DS_Store
 Thumbs.db
+
+# Firebase
+.firebase
+*-debug.log
+.runtimeconfig.json
```
Comment (tatianahutii) [inline frontend/.gitignore:47]: Why did you add this?

File: frontend/src/app/app.module.ts [hunk 12]
```diff
@@ -18,11 +23,14 @@ import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
         AppRoutingModule,
         NgbModule,
         BrowserAnimationsModule,
+        provideFirebaseApp(() => initializeApp(environment.firebaseConfig)),
+        provideAuth(() => getAuth()),
```
Comment (tatianahutii) [inline frontend/src/app/app.module.ts:27]: Do we need this?

File: frontend/src/app/core/interceptors/jwt.interceptor.ts [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+@Injectable()
+export class JwtInterceptor implements HttpInterceptor {
+
+  constructor(private authService: AuthService) {}
+
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+
+    const currentUser = this.authService.currentUser$;
+    const accessToken = this.authService.token;
+    const isLoggedIn = currentUser && accessToken;
```
Comment (tatianahutii) [inline frontend/src/app/core/interceptors/jwt.interceptor.ts:20]: Why do you check a current user?

File: frontend/src/app/core/services/auth.service.ts [hunk 14]
```diff
@@ -0,0 +1,67 @@
+import {Injectable} from '@angular/core';
+import { from, switchMap} from "rxjs";
+import {
+  Auth,
+  authState,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signInWithPopup,
+  updateProfile,
+} from "@angular/fire/auth";
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts:10]: Please write this is one line

File: frontend/src/app/core/services/auth.service.ts [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+        user.getIdToken(true).then(theToken => {
+          this.token = theToken
+          localStorage.setItem('jwt', JSON.stringify(this.token));
+        })
+      }
+      else {
+        localStorage.removeItem('jwt');
+        this.token = null;
+      }
+    })
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts:33]: Do we need this?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/Codi.Core/Codi.Core.WebAPI/Codi.Core.WebAPI.csproj (hunk 10 above)
Comment (SayPls) [reply backend/Codi.Core/Codi.Core.WebAPI/Codi.Core.WebAPI.csproj:37]: no

File: frontend/src/app/app.module.ts (hunk 12 above)
Comment (SayPls) [reply frontend/src/app/app.module.ts:27]: yes, this is config for firebase

File: frontend/src/app/core/services/auth.service.ts (hunk 14 above)
Comment (SayPls) [reply frontend/src/app/core/services/auth.service.ts:10]: ok

File: frontend/src/app/core/services/auth.service.ts (hunk 15 above)
Comment (SayPls) [reply frontend/src/app/core/services/auth.service.ts:33]: if the token will be stored in a variable, then we don't need storage

Comment (tatianahutii) [reply frontend/src/app/core/services/auth.service.ts]: Why? You will need to get it from storage after every page reload

File: frontend/src/app/core/interceptors/jwt.interceptor.ts (hunk 13 above)
Comment (tatianahutii) [reply frontend/src/app/core/interceptors/jwt.interceptor.ts:20]: I believe it would be better use just token

**Review by ismv — APPROVED** [review]

**Review by ismv — APPROVED** [review]

**Pull request merged**